app/scripts/controllers/transactions/index.js

```diff
@@ -366,7 +366,42 @@ class TransactionController extends EventEmitter {
     this.txStateManager.setTxStatusSubmitted(txId)
   }
 
-  confirmTransaction (txId) {
+  /**
+    Sets the status of the transaction to confirmed
+    and sets the status of nonce duplicates as dropped
+    if the txParams have data it will fetch the txReceipt
+    @param txId {number} - the tx's Id
+    @returns {Promise<void>}
+  */
+
+  async confirmTransaction (txId) {
+    // get the txReceipt before marking the transaction confirmed
+    // to ensure the receipt is gotten before the ui revives the tx
+    const txMeta = this.txStateManager.getTx(txId)
+
+    if (!txMeta) {
+      return
+    }
+
+    try {
+      const txReceipt = await this.query.getTransactionReceipt(txMeta.hash)
+
+      // It seems that sometimes the numerical values being returned from
+      // this.query.getTransactionReceipt are BN instances and not strings.
+      const gasUsed = txReceipt.gasUsed instanceof ethUtil.BN
```

Comment:
I think it might be safer to check `typeof txReceipt.gasUsed !== 'string'` here instead if that works (JS is weird with it's `instanceof` and how modules often work such that this could be false for some instances of BN)